packages/ui-contract-editor/src/lib/components/Conditional/index.js

```diff
@@ -27,32 +30,21 @@ const Conditional = React.forwardRef((props, ref) => {
   const [hovering, setHovering] = useState(false);
   const conditionalPath = ReactEditor.findPath(editor, node);
   const isNotReadOnly = !props.readOnly;
-  const isNotEmptyString = node.children[0].text !== '';
-  const isEmptyString = node.children[0].text === '';
-  const conditional = {
-    id: data.id,
-    whenTrue: data.whenTrue,
-    whenFalse: data.whenFalse,
-    isFalse: (
-      node.children[0].text === data.whenFalse
-    ),
-  };
+  const isContentShowing = data.isTrue
+    ? !!childReducer(data.whenTrue).length
+    : !!childReducer(data.whenFalse).length;
 
   const toggleConditional = (path) => {
     const newConditional = {
       object: 'inline',
       type: CONDITIONAL,
       data: {
-        id: conditional.id,
-        whenTrue: conditional.whenTrue,
-        whenFalse: conditional.whenFalse
+        name: data.name,
+        whenTrue: data.whenTrue,
+        whenFalse: data.whenFalse,
+        isTrue: !data.isTrue,
       },
-      children: [{
-        object: 'text',
-        text: conditional.isFalse
-          ? conditional.whenTrue
-          : conditional.whenFalse
-      }]
+      children: data.isTrue ? data.whenFalse : data.whenTrue
```

Comment:
That looks backward?

Reply:
Yeah this is creating the opposite of what currently exists in order to insert it. So if it `isTrue`, we want to insert the `whenFalse` value to switch it on click.